transports/tcp/src/lib.rs

```diff
@@ -78,10 +72,10 @@ pub struct TcpConfig {
     nodelay: Option<bool>,
 }
 
-impl TcpConfig {
+impl Default for TcpConfig {
     /// Creates a new configuration object for TCP/IP.
     #[inline]
-    pub fn new() -> TcpConfig {
```

Comment:
I'm not a fan of removing `new()` entirely.

Reply:
This was primarily motivated because `TcpConfig::new()` and `TcpConfig::default()` produced differently configured values and I wanted to avoid the ambiguity. Also I thought of `TcpConfig` really only as a configuration object, where `default` seemed more natural sounding. But I can define one in terms of the other if you prefer.